snuba/subscriptions/data.py

```diff
@@ -172,7 +167,13 @@ def build_request(
     def from_dict(
         cls, data: Mapping[str, Any], entity_key: EntityKey
     ) -> SubscriptionData:
-        entity_subscription = get_entity_subscription(entity_key)(data_dict=data)
+
+        entity_subscription = get_entity(entity_key).get_entity_subscription()
+        if not entity_subscription:
+            raise InvalidSubscriptionError(
+                f"entity subscription for {entity_key} does not exist"
+            )
+        entity_subscription = entity_subscription.set_org(data_dict=data)
```

Comment:
The org should be parsed out of the data here, instead of inside the `set_org` function.